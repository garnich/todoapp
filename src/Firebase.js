import firebase from 'firebase/app';
import 'firebase/database';
import 'firebase/auth';
import firebaseConfig from '../config';
 
firebase.initializeApp(firebaseConfig);

export const provider = new firebase.auth.GoogleAuthProvider();
export const auth = firebase.auth();  
const msgStyle = 'background: #000; color: green; font-size: 14px; font-weight: 500';
export const errorCatcher = (error, msg) => {
  error ? console.error(error) : console.log(`%c ${msg} SUCSESSFULY!`, msgStyle)
};
const currentDate = new Date().toISOString();

firebase.database.enableLogging(errorCatcher(false, `IN ${currentDate} connectinf to [FIREBASE]: `));

export default firebase;