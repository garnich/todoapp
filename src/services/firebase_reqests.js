import firebase, { errorCatcher } from './Firebase'

const updateDataInFirebase = (uid, data, msg) => {
    firebase
      .database()
      .ref(`${uid}/todo`)
      .set(data, error => errorCatcher(error, msg))
}

const updateItemInFireBase = (itemIndex, newItemData, msg, uid) => {

    firebase
      .database()
      .ref(`${uid}/todo/${itemIndex}`)
      .update(newItemData, error => errorCatcher(error, msg))
}

const deleteItemInFireBase = (data, msg, uid) => {

    firebase
      .database()
      .ref(`${uid}/todo`)
      .remove()

      updateDataInFirebase(uid, data, msg)
}

const dataRefFirebase = (uid, func, msg) => { 
    firebase
        .database()
        .ref(`${uid}/todo`)
        .on('value', snapshot => {
            const data = snapshot.val()

            func(data),
            error => errorCatcher(error, msg)
        })
}


export  { 
    updateDataInFirebase, 
    updateItemInFireBase, 
    deleteItemInFireBase,
    dataRefFirebase
}