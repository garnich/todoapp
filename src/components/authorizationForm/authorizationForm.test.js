import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import AuthorizationForm from './authorizationForm'
import { auth } from './../../services/Firebase'

Enzyme.configure({ adapter: new Adapter() })

describe('Test AuthorizationForm component', () => {
  xit('<AuthorizationForm> test snapshot', () => {
    const wrapper = shallow(<AuthorizationForm />)

    expect(wrapper).toMatchSnapshot()
  })

  xit('<AuthorizationForm> handleInputChange test', () => {
    const event = {
      target: {
        name: 'singInEmail',
        value: 'Test Value',
      },
    }
    const wrapper = shallow(<AuthorizationForm />)

    wrapper.instance().handleInputChange(event)

    expect(wrapper.state('singInEmail')).toEqual('Test Value')
  })

  xit('<AuthorizationForm> handleSingIn test FAILED auth PASSWORD', async () => {
    const event = {
      target: [{ value: 'test@test.test' }, { value: '' }],
      preventDefault: jest.fn(),
    }
    const error =
      'The password is invalid or the user does not have a password.'
    const wrapper = shallow(<AuthorizationForm />)

    wrapper.instance().handleSignIn(event)

    await expect(wrapper.state('createUserWithEmailAndPassword')).toEqual(false)
    await expect(wrapper.state().error.message).toEqual(error)
  })

  xit('<AuthorizationForm> handleSingnUp test', () => {
    const event = {
      target: [
        { value: 'test@test.test' },
        { value: 'testPass' },
        { value: 'testPass' },
      ],
      preventDefault: jest.fn(),
    }

    const wrapper = shallow(<AuthorizationForm />)

    expect(wrapper.state('createUserWithEmailAndPassword')).toEqual(false)

    auth.createUserWithEmailAndPassword = jest.fn((para1, para2) => {
      return new Promise(resolve =>
        resolve(
          wrapper.setState({
            createUserWithEmailAndPassword: true,
          })
        )
      )
    })

    wrapper.instance().handleSignUp(event)
    expect(wrapper.state('createUserWithEmailAndPassword')).toEqual(true)
  })

  xit('<AuthorizationForm> handleSingnUp test with different PASSWORDS', () => {
    const event = {
      target: [
        { value: 'test@test.test' },
        { value: 'testPass' },
        { value: 'testPass1' },
      ],
      preventDefault: jest.fn(),
    }

    const wrapper = shallow(<AuthorizationForm />)

    wrapper.instance().handleSignUp(event)

    expect(wrapper.state().error.message).toEqual('Passwords must be same')
  })
})
