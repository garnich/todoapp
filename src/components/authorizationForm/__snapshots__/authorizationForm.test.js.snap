// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AuthorizationForm component <AuthorizationForm> test snapshot 1`] = `
<div
  className="errorWrapper"
>
  <div
    className="loginErrorWrapper"
  />
  <div
    className="authWrapper"
  >
    <form
      autoComplete="off"
      className="col-3"
      onSubmit={[Function]}
      role="form"
    >
      <h2
        className="title"
      >
        SignIn
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="email1"
        >
          Email address:
        </label>
        <input
          className="form-control"
          id="email1"
          name="singInEmail"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="pwd"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="pwd"
          name="singInPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <form
      autoComplete="off"
      className="col-3"
      onSubmit={[Function]}
      role="form"
    >
      <h2
        className="title"
      >
        SignUp
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="email2"
        >
          Email address:
        </label>
        <input
          className="form-control"
          id="email2"
          name="singUpEmail"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="pwd1"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="pwd1"
          name="singUpPassword1"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="pwd2"
        >
          Confirm password:
        </label>
        <input
          className="form-control"
          id="pwd2"
          name="singUpPassword2"
          onChange={[Function]}
          placeholder="Repeat password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
